tests/test_cli.py

```diff
@@ -332,3 +333,88 @@ def _validate_intermediate_output(*args):
         server_b_published_output_filename,
     ):
         assert json.load(open(filename)) == [3, 2, 1]
+
+
+def test_verify1_ignores_bad_payloads_in_batch(tmp_path, shared_seed):
+    key_a = _keygen()
+    key_b = _keygen()
+    key_c = _keygen()
+
+    batch_id = "test"
+    n_data = 3
+    base_args = ["--n-data", n_data, "--batch-id", batch_id]
+
+    runner = CliRunner()
+
+    # create dependencies for encode function
+    data_path = tmp_path / "data.json"
+    with open(data_path, "w") as f:
+        f.write(json.dumps([1, 0, 1]))
+    ignore = tmp_path / "ignore"
+    ignore.mkdir()
+
+    def encode(key_0, key_1, output_path):
+        args = base_args + [
+            "--public-key-hex-internal",
+            key_0["public_key"],
+            "--public-key-hex-external",
+            key_1["public_key"],
+            "--input",
+            data_path,
+            "--output-A",
+            output_path,
+            "--output-B",
+            ignore,
+        ]
+        result = runner.invoke(commands.encode_shares, args)
+        assert result.exit_code == 0
+
+    # generate two sets of data points with different pairs of keys
+    out_0 = tmp_path / "out-0"
+    out_1 = tmp_path / "out-1"
+    out_0.mkdir()
+    out_1.mkdir()
+
+    out_full = tmp_path / "out.json"
+    out_verify1 = tmp_path / "verify1"
+
+    encode(key_a, key_c, out_0)
+    encode(key_b, key_c, out_1)
```

Comment:
I know this is only a test, but some of this could be expressed much more concisely. e.g.

```py
(out_0, out_1) = [tmp_path / f"out-{i}" for i in range(2)]
for (key_1, key_2, out_dir) in [(key_a, key_c, out_0), (key_b, key_c, out_1)]:
    encode(key_1, key_2, out_dir)
out_full = tmp_path / "out.json"
out_verify1 = tmp_path / "verify1"
```

Reply:
I find the second line difficult to read, but I do agree that the test is quite long.